src/main/kotlin/org/rust/ide/hints/RsExpressionTypeProvider.kt

```diff
@@ -35,7 +36,7 @@ class RsExpressionTypeProvider : ExpressionTypeProvider<PsiElement>() {
 
     override fun getInformationHint(element: PsiElement): String {
         val type = getType(element)
-        return type.toString().escaped
+        return type.insertionSafeTextWithAliases.escaped
```

Comment:
You should use `textWithAliasNames` instead of `insertionSafeTextWithAliases ` (I know that it's marked a `TestOnly`. It's time to drop this annotation) not to change the previous style of how we render unknown types. `insertionSafeTextWithAliases` is used as text for PSI element creation in intentions/quick fixes/refactorings/etc.